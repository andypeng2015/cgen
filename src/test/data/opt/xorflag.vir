module foo

export function w $foo(w %x) {
@start:
  %c = lt.w %x, 5_w
  %f = flag.w %c
  %f = xor.w %f, 1_w
  ret %f
}
