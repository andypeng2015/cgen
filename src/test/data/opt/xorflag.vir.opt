module foo

export function w $foo(w %x) {
@2:
  %2 = ge.w %x, 0x5_w ; @8
  %3 = flag.w %2 ; @9
  ret %3
}
