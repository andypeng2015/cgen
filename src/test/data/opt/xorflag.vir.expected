module foo

export function w $foo(w %x) {
@0:
  %1 = ge.w %x, 0x5_w ; @5
  %0 = flag.w %1 ; @4
  ret %0
}
